test/spec/calculate.spec.js

```diff
@@ -146,9 +214,662 @@ describe('calculate functionality', () => {
+            );
+
+            feverDurationDayUnit.checked = true;
+            feverDurationDayUnit.dispatchEvent(events.Change());
+
+            timers.runAll();
+
+            const feverDurationDays = form.view.html.querySelector(
+                '[name="/data/consented/illhistory/signs_symptoms_final_illness/Id10148_b"]'
+            );
+
+            feverDurationDays.value = '36';
+            feverDurationDays.dispatchEvent(events.Change());
+
+            timers.runAll();
+
+            const isDateOfBirthKnownModelNode = form.model
+                .node(isDateOfBirthKnown.name)
+                .getElement();
+            const dateOfBirthModelNode = form.model
+                .node(dateOfBirth.name)
+                .getElement();
+            const isDateOfDeathKnownModelNode = form.model
+                .node(isDateOfDeathKnown.name)
+                .getElement();
+            const yearOfDeathModelNode = form.model
+                .node(yearOfDeath.name)
+                .getElement();
+            const ageGroupModelNode = form.model
+                .node(ageGroup.name)
+                .getElement();
+            const didHaveFeverModelNode = form.model
+                .node(didHaveFever.name)
+                .getElement();
+            const feverDurationDayUnitModelNode = form.model
+                .node(feverDurationDayUnit.name)
+                .getElement();
+            const feverDurationDaysModelNode = form.model
+                .node(feverDurationDays.name)
+                .getElement();
+
+            expect(isDateOfBirthKnownModelNode.textContent).to.equal('yes');
+            expect(dateOfBirthModelNode.textContent).to.equal(
+                adultDateOfBirthValue
+            );
+            expect(isDateOfDeathKnownModelNode.textContent).to.equal('no');
+            expect(yearOfDeathModelNode.textContent).to.equal(yearOfDeathValue);
+            expect(ageGroupModelNode.textContent).to.equal('adult');
+            expect(didHaveFeverModelNode.textContent).to.equal('yes');
+            expect(feverDurationDayUnitModelNode.textContent).to.equal('days');
+            expect(feverDurationDaysModelNode.textContent).to.equal('36');
+
+            didHaveFever.checked = false;
+            didHaveFever.dispatchEvent(events.Change());
+
+            timers.runAll();
+
+            expect(feverDurationDayUnitModelNode.textContent).to.equal('');
+            expect(feverDurationDaysModelNode.textContent).to.equal('');
+
+            didHaveFever.checked = true;
+            didHaveFever.dispatchEvent(events.Change());
+
+            timers.runAll();
+
+            expect(feverDurationDayUnitModelNode.textContent).to.equal('days');
+            expect(feverDurationDaysModelNode.textContent).to.equal('36');
+        }, 1000000);
+
+        // This test, as with the larger one above, should be moved to a relevance-specific suite.
+        // But I am le tired.
```

Comment:
Still le tired?

An alternative would be to pull out tests on this form into their own world. In JR it has acted as a general smoketest because there's so much going on in it.